Build failure: xzoom

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix build nixpkgs/634fd46801442d760e09493a794c4f15db2d0cbb#xzoom` (note: identical result with `nix build -option sandbox true`.)

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/5pl0jnl2vmkdpmxqgf14ca0fk5zhnws7-xzoom-0.3.tgz
source root is xzoom-0.3
setting SOURCE_DATE_EPOCH to timestamp 833442891 of file xzoom-0.3/README
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/bgg8grcsm5y6cmkiggy5mcgrkbn63np4-xzoom_0.3-24.diff.gz
patching file Imakefile
patching file debian/changelog
patching file debian/compat
patching file debian/control
patching file debian/copyright
patching file debian/install
patching file debian/menu
patching file debian/rules
patching file debian/watch
patching file debian/xzoom.desktop
patching file scale.h
patching file xzoom.c
patching file xzoom.man
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring with imake: ''
imake -DUseInstalled -I/nix/store/8b7krzkp8f7d58vw7sc82xids0k77g8j-imake-1.0.10/lib/X11/config
make Makefiles
make: Nothing to be done for 'Makefiles'.
make includes
make: Nothing to be done for 'includes'.
make depend
gccmakedep  --   -I/nix/store/qqhl2a7166iik31sr9jklf3m9gfyjlrq-xorg-cf-files-1.0.8/include    -Dlinux -D__amd64__ -D_POSIX_C_SOURCE=199309L 				-D_POSIX_SOURCE -D_XOPEN_SOURCE 				-D_BSD_SOURCE -D_SVID_SOURCE                                 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 				  				  -DFUNCPROTO=15 -DNARROWPROTO   -DFRAME -DXSHM   -- xzoom.c
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash PREFIX=\$\(out\) BINDIR=\$\(out\)/bin MANPATH=\$\(out\)/share/man
gcc -O2 -fno-strength-reduce -fno-strict-aliasing      -I/nix/store/qqhl2a7166iik31sr9jklf3m9gfyjlrq-xorg-cf-files-1.0.8/include    -Dlinux -D__amd64__ -D_POSIX_C_SOURCE=199309L 				-D_POSIX_SOURCE -D_XOPEN_SOURCE 				-D_BSD_SOURCE -D_SVID_SOURCE                                 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 				  				  -DFUNCPROTO=15 -DNARROWPROTO   -DFRAME -DXSHM    -c -o xzoom.o xzoom.c
In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:28,
                 from xzoom.c:33:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/features.h:197:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
  197 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
      |   ^~~~~~~
xzoom.c: In function 'main':
xzoom.c:620:33: warning: 'XKeycodeToKeysym' is deprecated [-Wdeprecated-declarations]
  620 |                                 switch(XKeycodeToKeysym(dpy, event.xkey.keycode, 0)) {
      |                                 ^~~~~~
In file included from xzoom.c:39:
/nix/store/2p1gkzwz2lq6vv67579vh4bllw1psqk8-libX11-1.8.10-dev/include/X11/Xlib.h:1683:15: note: declared here
 1683 | extern KeySym XKeycodeToKeysym(
      |               ^~~~~~~~~~~~~~~~
xzoom.c:629:33: warning: 'XKeycodeToKeysym' is deprecated [-Wdeprecated-declarations]
  629 |                                 switch(XKeycodeToKeysym(dpy, event.xkey.keycode, 0)) {
      |                                 ^~~~~~
/nix/store/2p1gkzwz2lq6vv67579vh4bllw1psqk8-libX11-1.8.10-dev/include/X11/Xlib.h:1683:15: note: declared here
 1683 | extern KeySym XKeycodeToKeysym(
      |               ^~~~~~~~~~~~~~~~
xzoom.c:756:41: error: implicit declaration of function 'alarm' [-Wimplicit-function-declaration]
  756 |                                         alarm(2);
      |                                         ^~~~~
xzoom.c:888:25: error: implicit declaration of function 'usleep' [-Wimplicit-function-declaration]
  888 |                         usleep(delay);
      |                         ^~~~~~
make: *** [<builtin>: xzoom.o] Error 1

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.0-50-generic, Ubuntu, 24.04.1 LTS (Noble Numbat), nobuild`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@7c6f434c 
---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/xz/xzoom/package.nix
+++ b/pkgs/by-name/xz/xzoom/package.nix
@@ -26,6 +26,10 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  postPatch = ''
+    sed -i 1i'#include <unistd.h>' xzoom.c
+  '';
+
   nativeBuildInputs = [
     imake
     gccmakedep

